File: LeetCode_3201/Main.js

/**
 * @param {number[]} nums
 * @return {number}
 */

function max(a, b) {
    return a > b ? a : b;
}

var maximumLength = function(nums) {
    let len = 0;
    let mod = 0;
    let odd = 0;

    for (let i = 0; i < nums.length; i++) {
        let x = nums[i];
        if (x % 2 != mod) {
            len += 1;
            mod = x & 1;
        }
        odd += (x & 1);
    }

    let mx = max(len, odd);
    mx = max(mx, nums.length - odd);

    len = 0, mod = 1;
    
    for (let i = 0; i < nums.length; i++) {
        let x = nums[i];
        if (x % 2 != mod) {
            len += 1;
            mod = x & 1;
        }
    }

    return max(mx, len);
};

File: LeetCode_3201/Main.c

int max(int x, int y) {
    return x > y ? x : y;
}

int maximumLength(int* nums, int numsSize) {
    int len = 0;
    int mod = 0;
    int odd = 0;

    for (int i = 0; i < numsSize; i++) {
        int x = nums[i];
        if (x % 2 != mod) {
            len += 1;
            mod = x & 1;
        }
        odd += (x & 1);
    }

    int mx = max(len, odd);
    mx = max(mx, numsSize - odd);

    len = 0, mod = 1;
    
    for (int i = 0; i < numsSize; i++) {
        int x = nums[i];
        if (x % 2 != mod) {
            len += 1;
            mod = x & 1;
        }
    }

    return max(mx, len);
}